x-pack/plugins/apm/server/routes/event_metadata/get_event_metadata.ts

```diff
@@ -18,43 +18,26 @@ export async function getEventMetadata({
   apmEventClient,
   processorEvent,
   id,
+  start,
+  end,
 }: {
   apmEventClient: APMEventClient;
   processorEvent: ProcessorEvent;
   id: string;
+  start: number;
+  end: number;
 }) {
-  const filter: QueryDslQueryContainer[] = [];
-
-  switch (processorEvent) {
-    case ProcessorEvent.error:
-      filter.push({
-        term: { [ERROR_ID]: id },
-      });
-      break;
-
-    case ProcessorEvent.transaction:
-      filter.push({
-        term: {
-          [TRANSACTION_ID]: id,
-        },
-      });
-      break;
-
-    case ProcessorEvent.span:
-      filter.push({
-        term: { [SPAN_ID]: id },
-      });
-      break;
-  }
-
+  const fieldName = getFieldName(processorEvent);
   const response = await apmEventClient.search('get_event_metadata', {
     apm: {
       events: [processorEvent],
     },
     body: {
       track_total_hits: false,
       query: {
-        bool: { filter },
+        bool: {
+          filter: [...rangeQuery(start, end), { term: { [fieldName]: id } }],
```

Comment:
So the start and end date is going to be the same in this case (both using `@timestamp`).